### ISSUE TITLE
PassFF doesn't work with Firefox 71

### ISSUE THREAD

**Issue by mhellwig** [passff/passff#427]
### General information. Describe your environment

- Versions
  - Operating system: Fedora Desktop 31
  - Browser: Firefox 71.0
  - PassFF: 1.9.1
  - Host app: 1.2.1

Status line output: `there is nothing ..`

- Output of `pass`:  ... long list of my accounts ...

---

### Actual behaviour
I _think_ nothing was updated today (neither Firefox nor pass nor passff), which makes this ultra weird. BUT. As of today (before, it worked) the PassFF extension does nothing. Meaning login fields no longer have the P icon in the right corner and if I click on the P icon in the menubar, I get a completely empty dropdown. The buttons "+", "refresh" and "settings" still exist but don't cause anything to happen when clicked. pass in a terminal still works as expected. The "check the output of the host app" step in the troubleshooting guide delivers expected results.

Still, completely empty dropdown, no p-icons. Any ideas?

**Comment by mhellwig** [comment]
addendum: other extensions work (uBlock0, Vimium, Dark Reader)

**Comment by Larifantari** [comment]
Can confirm this issue, same setup and same problem at my side.

**Comment by TurboGit** [comment]
Likewise. On my side  Firefox was updated from 70 to 71 and now PassFF is not working (empty window). I'm on Debian/sid.

**Comment by mowgli** [comment]
Let me add my metoo.

Debian sid update from 70 to 71 and passff stop working.

I also am not able to refresh firefox as I will loose that many tabs and even essential plugins without firefox is unusable. Also many securety settings will get lost.

**Comment by mowgli** [comment]
By the way, I get the following two messages in browser console when I open the dropdown.
```
sender.tab is undefined 2 MessageChannel.jsm:1019
    promise resource://gre/modules/MessageChannel.jsm:1019
ConstraintError: A mutation operation in the transaction failed because a constraint was not satisfied. ExtensionStorageIDB.jsm:812
    normalizeStorageError resource://gre/modules/ExtensionStorageIDB.jsm:812
    method chrome://extensions/content/child/ext-storage.js:273
    InterpretGeneratorResume self-hosted:1159
    AsyncFunctionThrow self-hosted:695
```

**Comment by tuxor1337** [comment]
This is (just as #425) possibly caused by the upstream bug in Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1601424) or even in gcc (https://bugzilla.mozilla.org/show_bug.cgi?id=1601707). I don't think, we can't do anything about this, before they have sorted out the upstream issues.

**Comment by psukys** [comment]
The reporter in Mozilla's bugzilla has identified a good distinction that distro installed Firefox has this defect, whereas if Firefox is downloaded from their website (tar.bz2 archive), PassFF works without any hiccups, which I checked as well and it seems to check out.

**Comment by mhellwig** [comment]
can confirm, downgrading to 69 on fedora solves the problem, using the 71 from the firefox webisite (not built by Fedora) also solves the problem. So actually a distro problem, in a way?

**Comment by MeisterP** [comment]
> This is (just as #425) possibly caused by the upstream bug in Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1601424) or even in gcc (https://bugzilla.mozilla.org/show_bug.cgi?id=1601707).

If I compile `firefox-71.0` with `gcc-9.2.0`, passff doesn't work. Compiling `firefox-71.0` with `clang-8.0.1` and passff works without any issue. I'm using gentoo.



> So actually a distro problem, in a way?

Yes. It depends on what compiler your distribution uses to create the firefox package.

**Comment by psteinb** [comment]
same here with Fedora `fc30` and kernel `5.3.14-200.fc30.x86_64`

**Comment by xundeenergie** [comment]
same here fedora fc31 with firefox 71

**Comment by brother** [comment]
As a workaround for Debian I downgraded to Firefox 70 via snapshots.

Download the [deb](https://snapshot.debian.org/package/firefox/70.0.1-1/#firefox_70.0.1-1:2b:b1) and install it `sudo apt install ./path/to/irefox_70.0.1-1+b1_amd64.deb` - when starting firefox use the switch `--allow-downgrade` or else it will force quit until you create a new profile.

**Comment by tuxor1337** [comment]
There are newer builds of Firefox 71 available for Fedora users that fix this issue: 
https://koji.fedoraproject.org/koji/buildinfo?buildID=1420744 (fc30)
https://koji.fedoraproject.org/koji/buildinfo?buildID=1420745 (fc31)
https://koji.fedoraproject.org/koji/buildinfo?buildID=1420743 (fc32)

**Comment by patrickrst** [comment]
Same for Debian, new build 71.0-2 (sid):
https://packages.debian.org/sid/firefox

**Comment by nning** [comment]
Fixed in Fedora 31 with `firefox-71.0-15`
